rails-questions.md

```diff
@@ -1,30 +1,32 @@
 # ASSESSMENT 5: Database and Active Record Practical Questions
 
 1. What is SQL?
+Structured Query Language, it is a domain specific language and communicates with relational databases.
 
-
-2. What the PostgreSQL query that would return all the content in a particular table?
-
+2. What is the PostgreSQL query that would return all the content in a particular table?
+SELECT *
+FROM table;
 
 3. What is the command to create a new Rails application with a PostgreSQL database?
-
+$ rails new app_name -d postgresql -T
 
 4. What is the command to generate a Rails model for a meals table with columns named breakfast, lunch, and dinner?
-
+$ rails g model Meal breakfast:string lunch:string dinner:string
 
 5. What is a migration? Why would you use one?
-
+Migration is used to migrate data from one or more source databases to another. When creating new columns for a database we need to migrate all of the new information to the main database and thats when we would use migration. We also migrate data when changing the table name, column name, data types removing columns, etc. All of those changes need to be updated by migration.
```

Comment:
Yea, so hen changing the database structure we use a migration, correct. This does not move the data around to other databases though. It's a weird nuance but it is important.